### PR TITLE
Proposing json-join again

### PR DESCRIPTION
**Pull request by xml-project**
Since we now have consensus, only to include those JSON-related steps, which can not be written in XPath, I propose this step again.
One can not write it in XPath because it relies on the order on the documents on the source port and fn:collection() does not preserve order.

### CODE REVIEW

File: step-json/src/main/xml/steps/join-json.xml [hunk 1]
```diff
@@ -6,15 +6,21 @@
...
+<title>p:json-join</title>
 
-<para>T.B.D.</para>
+<para>The <code>p:json-join</code> step joins a sequence of JSON documents into a single JSON document (an array).
+If the sequence on port <code>source</code> is empty, the empty sequence is returned on port <code>result</code>.</para>
 
 <p:declare-step type="p:join-json">
-  <p:input port="source"/>
-  <p:output port="result"/>
+  <p:input port="source" sequence="true" content-types="application/json application/*+json"> </p:input>
```
Comment (gimsieke) [inline step-json/src/main/xml/steps/join-json.xml]: Maybe use the `json` content type shortcut for input (and also output)?

File: step-json/src/main/xml/steps/join-json.xml [hunk 2]
```diff
@@ -6,15 +6,21 @@
...
+  <p:output port="result" content-types="application/json"/>
+  <p:option name="flatten-arrays" as="xs:boolean" select="false()" />
 </p:declare-step>
 
-<para>T.B.D.</para>
+<para>If option <code>flatten-arrays</code> is <literal>false</literal>, a member in the array is created for each document in
+the sequence appearing on port<code>source</code>. Therefor the produced array with have the same number of
```
Comment (gimsieke) [inline step-json/src/main/xml/steps/join-json.xml]: Therefore

File: step-json/src/main/xml/steps/join-json.xml [hunk 3]
```diff
@@ -6,15 +6,21 @@
...
+members as the number of documents appearing on port <code>source</code>. However if <code>flatten-arrays</code> is
+<literal>true</literal>, for each member of an array appearing at the top level of a JSON-document
```
Comment (gimsieke) [inline step-json/src/main/xml/steps/join-json.xml]: of a JSON document (without hyphen)

Comment (gimsieke) [inline step-json/src/main/xml/steps/join-json.xml]: of a JSON document (without hyphen)

**Comment by xml-project** [comment]
@gimsieke Somehow you commented on the first commit, but ignored the second in which the requested changes (expect shortcut which were introduced later) where already present.
For me it looks good now, but I am not sure because of your comment.
Merge?

**Comment by xml-project** [comment]
No idea, what went wrong here and why you see something other than me.
Will come back later with a new pull request for this.

**Pull request closed without merging**